lib/ui/highlight-manager.js

```diff
@@ -102,20 +169,23 @@ HighlightManager.prototype = {
       return;
     }
     this.recommendation.isHighlighted = true;
-    this.popup.selectedIndex = -1;
+    this.popup.el.selectedIndex = -1;
     this.win.requestAnimationFrame(() => {
       this.recommendation.isHighlighted = true;
-      this.popup.selectedIndex = -1;
+      this.popup.el.selectedIndex = -1;
       this.win.requestAnimationFrame(() => {
         this.recommendation.isHighlighted = true;
-        this.popup.selectedIndex = -1;
+        this.popup.el.selectedIndex = -1;
       });
     });
   },
   clearHighlight: function() {
-    const resultsContainer = this.win.document.getAnonymousElementByAttribute(this.popup, 'anonid', 'richlistbox');
-    const resultRows = resultsContainer.getElementsByClassName('autocomplete-richlistitem');
-    Array.prototype.forEach.call(resultRows, row => { row.selected = false; });
+    this.popup.el.richlistbox.selectedIndex = -1;
+    this.recommendation.isHighlighted = false;
+  },
+  clearRecommendationHighlight: function() {
+    // See attachMouseListeners() for documentation.
+    this.recommendation.isHighlighted = false;
```

Comment:
Seems funny to have a one-line method, but we need a non-anonymous wrapper so that we can unsubscribe the listener for the `'results-list-mouse-highlighted'` event.